loginterceptor/loginterceptor.go

```diff
@@ -104,3 +99,52 @@ func (i *PrefixInterceptor) run() {
 		i.logger.Errorf("router scanner error: %v\n", err)
 	}
 }
+
+// NonBlockingWriter is an io.Writer that writes to a wrapped io.Writer in a non-blocking way.
+type NonBlockingWriter struct {
+	channel chan string
+	wrapped io.Writer
+	logger  log.Logger
+}
+
+// NewNonBlockingWriter creates a new NonBlockingWriter.
+func NewNonBlockingWriter(w io.Writer, logger log.Logger) *NonBlockingWriter {
+	writer := &NonBlockingWriter{
+		channel: make(chan string, 100), // buffered channel to avoid blocking
+		wrapped: w,
+		logger:  logger,
+	}
+	go writer.Run()
+	return writer
+}
+
+// Write implements io.Writer. It writes into an internal pipe which the interceptor goroutine consumes.
+func (i *NonBlockingWriter) Write(p []byte) (int, error) {
+	select {
+	case i.channel <- string(p):
```

Comment:
let's not force string conversion, we can use a []byte channel to preserve the original format